grype/pkg/provider.go

```diff
@@ -4,18 +4,68 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/bmatcuk/doublestar/v2"
+
 	"github.com/anchore/stereoscope/pkg/image"
 	"github.com/anchore/syft/syft/source"
 )
 
 var errDoesNotProvide = fmt.Errorf("cannot provide packages from the given source")
 
 // Provide a set of packages and context metadata describing where they were sourced from.
-func Provide(userInput string, scopeOpt source.Scope, registryOptions *image.RegistryOptions) ([]Package, Context, error) {
+func Provide(userInput string, scopeOpt source.Scope, registryOptions *image.RegistryOptions, exclusions ...string) ([]Package, Context, error) {
 	packages, ctx, err := syftSBOMProvider(userInput)
 	if !errors.Is(err, errDoesNotProvide) {
+		if len(exclusions) > 0 {
+			packages, err = filterPackageExclusions(packages, exclusions)
+			if err != nil {
+				return nil, ctx, err
+			}
+		}
 		return packages, ctx, err
 	}
 
-	return syftProvider(userInput, scopeOpt, registryOptions)
+	return syftProvider(userInput, scopeOpt, registryOptions, exclusions...)
+}
+
+func filterPackageExclusions(packages []Package, exclusions []string) ([]Package, error) {
```

Comment:
nit: I see that this code is exercised via new integration tests, but the logic is complex enough here that unit tests for this specific function directly might not be a bad idea

Reply:
Added a number of new tests for the helper functions